src/gui/widgets/MidiCCRackView.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * MidiCCRackView.cpp - implementation of the MIDI CC rack widget
+ *
+ * Copyright (c) 2020 Ian Caio <iancaio_dev/at/hotmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+
+#include <QWidget>
+#include <QMdiSubWindow>
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QGridLayout>
+#include <QLabel>
+#include <QScrollArea>
+
+#include "MidiCCRackView.h"
+#include "GuiApplication.h"
+#include "MainWindow.h"
+#include "embed.h"
+#include "ComboBox.h"
+#include "ComboBoxModel.h"
+#include "GroupBox.h"
+#include "Knob.h"
+#include "Track.h"
+#include "InstrumentTrack.h"
+#include "TrackContainer.h"
+#include "BBTrackContainer.h"
+#include "Engine.h"
+#include "Song.h"
+#include "SongEditor.h"
+#include "BBEditor.h"
+
+
+MidiCCRackView::MidiCCRackView(InstrumentTrack * track) :
+	QWidget(),
+	m_track(track)
+{
+	setWindowIcon(embed::getIconPixmap("midi_cc_rack"));
+	setWindowTitle(tr("MIDI CC Rack"));
+
+	QMdiSubWindow * subWin = gui->mainWindow()->addWindowedWidget(this);
+
+	// Remove maximize button
+	Qt::WindowFlags flags = subWin->windowFlags();
+	flags &= ~Qt::WindowMaximizeButtonHint;
+	subWin->setWindowFlags(flags);
+
+	// Adjust window attributes, sizing and position
+	subWin->setAttribute(Qt::WA_DeleteOnClose, false);
+	subWin->move(
+		rand() % 731 + 50, // X is choosen randomly from 50 to 780
+		rand() % 331 + 50 // Y is choosen randomly from 50 to 380
```

Comment:
Why random?

Reply:
Though it was explained in Discord, just to keep it documented here: Position is set in a random range just so windows from multiple instrument's MIDI CC racks don't get created on top of each other.